tasks/create_project.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: "List existing projects"
+  uri:
+    url: "{{ gitlab_external_url }}/api/{{ gitlab_api_version }}/projects"
+    method: "GET"
+    headers:
+      PRIVATE-TOKEN: "{{ gitlab_admin_token }}"
+    validate_certs: "{{ gitlab_validate_certs }}"
+    return_content: True
+  register: projects_list_output
+
+- name: "Check if project is already exists"
+  set_fact:
+    project_exists: "{{ gitlab_project_group + '/' + gitlab_project_name in\
+      (projects_list_output.json | map(attribute='path_with_namespace') | list) }}"
+
+- name: "Create project"
+  uri:
+    url: "{{ gitlab_external_url }}/api/{{ gitlab_api_version }}/projects"
+    method: "POST"
+    headers:
+      PRIVATE-TOKEN: "{{ gitlab_admin_token }}"
+    body: "name={{ gitlab_project_name }}&namespace_id={{ gitlab_project_group_id }}"
+    body_format: raw
+    validate_certs: "{{ gitlab_validate_certs }}"
+    return_content: True
+    status_code: 201
+  register: project_output
+  when: not project_exists
+
+- name: "Set project ID"
+  set_fact:
+    gitlab_project_id: "{{ project_output.json.id }}"
+  when: not project_exists
+
+- name: "Get project ID"
+  set_fact:
+    gitlab_project_id: "{{ item.id }}"
+  loop: "{{ projects_list_output.json }}"
+  when:
+    - project_exists
+    - item.path_with_namespace == gitlab_project_group + '/' + gitlab_project_name
+
+- name: "Project debug"
```

Comment:
We still need it?

Reply:
yes - we need it to get the ID of created user, group and project for further needs